### PR TITLE
FEATURE: Discourse Gamification backfill scores

### PR DESCRIPTION
**Pull request by rngus2344**

### CODE REVIEW

**Comment by rngus2344** [comment]
'Recalculate Scores' button:
Screenshot 2023-07-18 at 12 09 21 PM

Recalculate range options:
Screenshot 2023-07-18 at 12 12 30 PM
Screenshot 2023-07-18 at 12 12 37 PM
Screenshot 2023-07-18 at 12 12 42 PM
Screenshot 2023-07-18 at 12 12 55 PM
Screenshot 2023-07-18 at 12 13 01 PM

For 'Custom Range', 'Apply' is not available when there is no input:
Screenshot 2023-07-18 at 12 13 06 PM
Screenshot 2023-07-18 at 12 13 14 PM

When you click 'Apply', daily recalculation remaining goes down and it tells admin when it's completed: 
Screenshot 2023-07-18 at 12 14 45 PM

When daily recalculation remaining is 0, 'Apply' button is not available:
Screenshot 2023-07-18 at 12 15 14 PM
Screenshot 2023-07-18 at 12 15 07 PM

File: jobs/regular/recalculate_scores.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Jobs
+  class RecalculateScores < ::Jobs::Base
+    def execute(opts = { since: "2014-8-26" })
```
Comment (xfalcox) [inline jobs/regular/recalculate_scores.rb:5]: Why this date?

And if we need something, maybe default to 10.days.ago like the default?

**Review by xfalcox — CHANGES_REQUESTED** [review]
Please change the default job date.

**Comment by xfalcox** [comment]
@ellaestigoy can you please give a pass on the modal ? I feel like spacing is a bit off, specially with the date.

File: jobs/regular/recalculate_scores.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      DiscourseGamification::GamificationScore.calculate_scores(since_date: opts[:since])
+
+      ::MessageBus.publish "/recalculate_scores",
```
Comment (xfalcox) [inline jobs/regular/recalculate_scores.rb:8]: Can this message be sent only to the user who started the recalculation?

**Comment by ellaestigoy** [comment]
> @ellaestigoy can you please give a pass on the modal ? I feel like spacing is a bit off, specially with the date.

Agreed, I'll check it locally first.

File: assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  <:body>
+    {{#if (eq this.status "loading")}}
+      <div class="status">{{i18n "gamification.recalculating"}}</div>
+    {{else if (eq this.status "complete")}}
+      <div class="status">{{d-icon "check"}}
+        {{i18n "gamification.completed"}}</div>
+    {{else}}
+      <form class="form-horizontal">
+        <div class="input-group">
+          <div>{{i18n "gamification.update_scores_help"}}</div>
```
Comment (ellaestigoy) [inline assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs:14]: Can we add a class name here, `recalculate-modal__lead-text` (we use BEM for naming) so we can target it for styling.

File: assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+              <DatePickerPast
+                @id="custom-from-date"
+                @class="date-input"
+                @placeholder="yyyy-mm-dd"
+                @value={{recalculateFromDate}}
+                @onSelect={{action (mut recalculateFromDate)}}
+              />
+            </div>
+          {{else}}
+            <div class="date-range">
```
Comment (ellaestigoy) [inline assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs:34]: Can we rename it to `recalculate-modal__date-range` (we use BEM for naming).

File: assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      @action={{@closeModal}}
+      @class="btn-primary"
+      @label={{if
+        (eq status "complete")
+        "gamification.close"
+        "gamification.cancel"
+      }}
+      @ariaLabel="gamification.cancel"
+    />
+    <div class="remaining">{{remainingText}}</div>
```
Comment (ellaestigoy) [inline assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs:62]: 💅🏼 Can we rename it to `recalculate-modal__footer-text`

**Review by ellaestigoy — DISMISSED** [review]

File: assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs [hunk 6]
```diff
@@ -0,0 +1,64 @@
+<DModal
+  class="recalculate-scores-form-modal"
+  @title={{i18n "gamification.recalculate"}}
+>
+  <:body>
+    {{#if (eq this.status "loading")}}
+      <div class="status">{{i18n "gamification.recalculating"}}</div>
+    {{else if (eq this.status "complete")}}
+      <div class="status">{{d-icon "check"}}
```
Comment (ellaestigoy) [inline assets/javascripts/discourse/components/modal/recalculate-scores-form.hbs:9]: 💅🏼 Can we rename it to `recalculate-modal__status`?

File: config/locales/client.en.yml [hunk 7]
```diff
@@ -39,10 +39,27 @@ en:
...
       cancel: "Cancel"
+      close: "Close"
       delete: "Delete"
       edit: "Edit"
       back: "Back"
       save: "Save"
+      apply: "Apply"
+      recalculate: "Recalculate Scores"
+      recalculating: "Recalculating Scores..."
+      completed: "Successfully done!"
```
Comment (ellaestigoy) [inline config/locales/client.en.yml:50]: Let's make the copy clearer, "Done! The scores have been successfully recalculated."

**Comment by rngus2344** [comment]
Okay, I made those changes! Please let me know if there's any other issues!

**Review by xfalcox — APPROVED** [review]

**Pull request merged**